File: electronic-station/brackets.py
def checkio(x):
	m={'(':')','{':'}','[':']'}
	s=[]
	for e in x:
		if e in'({[':s+=[m[e]]
		if e in')}]'and(not s or s.pop()!=e):return False
	return not s

if False:
	import sys
	if sys.version_info[0]>=3: raw_input=input
	n=int(raw_input())
	for _ in range(n):
		print(['NO','YES'][checkio(raw_input().strip())])
else:
	assert checkio("((5+3)*2+1)") == True, "Simple"
	assert checkio("{[(3+1)+2]+}") == True, "Different types"
	assert checkio("(3+{1-1)}") == False, ") is alone inside {}"
	assert checkio("[1+1]+(2*2)-{3/3}") == True, "Different operators"
	assert checkio("(({[(((1)-2)+3)-3]/3}-3)") == False, "One is redundant"

File: electronic-station/brackets.js
"use strict";

var brackets=function(x){
	var m={'(':')','{':'}','[':']'};
	var s=[];
	for(var e of x){
		if('({['.indexOf(e)>=0)s.push(m[e]);
		if(')}]'.indexOf(e)>=0 && (!s.length||s.pop()!=e))return false;
	}
    return !s.length;
};

var assert = require('assert');

if (!global.is_checking) {
	assert.equal(brackets("((5+3)*2+1)"), true, "Simple");
	assert.equal(brackets("{[(3+1)+2]+}"), true, "Different types");
	assert.equal(brackets("(3+{1-1)}"), false, ") is alone inside {}");
	assert.equal(brackets("[1+1]+(2*2)-{3/3}"), true, "Different operators");
	assert.equal(brackets("(({[(((1)-2)+3)-3]/3}-3)"), false, "One is redundant");
	assert.equal(brackets("2+3"), true, "No brackets, no problem");
	console.log("Coding complete? Click 'Check' to review your tests and earn cool rewards!");
}